server/src-lib/Hasura/GraphQL/Resolve/InputValue.hs

```diff
@@ -121,11 +122,27 @@ parseMany fn v = case v of
   AGArray _ arrM -> mapM (mapM fn) arrM
   _              -> tyMismatch "array" v
 
+onlyText
+  :: (MonadError QErr m)
+  => PGColValue -> m Text
+onlyText = \case
+  PGValText t -> return t
+  PGValVarchar t -> return t
+  _           -> throw500 "expecting text for onlyText"
+
 asPGColText
   :: (MonadError QErr m)
   => AnnGValue -> m Text
 asPGColText val = do
   (_, pgColVal) <- asPGColVal val
   case pgColVal of
```

Comment:
You can do
```haskell
onlyText pgColVal
```
Redundant logic found in `asPGColVal` and `onlyText`. Use `onlyText` in `asPGColVal` to replace pattern match.